### PR TITLE
1.19.3 resource loader changes

### PR DESCRIPTION
**Pull request by apple502j**
Fixes various issues with 22w42a porting and other linter yelling stuff. Should also make "This world is experimental" warning disappear for modded worlds, as mods aren't experimental.

### CODE REVIEW

File: fabric-resource-loader-v0/src/client/java/net/fabricmc/fabric/mixin/resource/loader/client/VanillaResourcePackProviderMixin.java [hunk 1]
```diff
@@ -23,16 +23,20 @@
...
-import net.minecraft.client.resource.ServerResourcePackProvider;
+import net.minecraft.client.resource.DefaultClientResourcePackProvider;
 import net.minecraft.resource.ResourcePackProfile;
+import net.minecraft.resource.VanillaResourcePackProvider;
 
 import net.fabricmc.fabric.impl.resource.loader.ModResourcePackCreator;
 
-@Mixin(ServerResourcePackProvider.class)
-public class ServerResourcePackProviderMixin {
+@Mixin(VanillaResourcePackProvider.class)
```
Comment (Technici4n) [inline fabric-resource-loader-v0/src/client/java/net/fabricmc/fabric/mixin/resource/loader/client/VanillaResourcePackProviderMixin.java:32]: What does this change?

File: fabric-resource-loader-v0/src/main/java/net/fabricmc/fabric/impl/resource/loader/FabricModResourcePack.java [hunk 2]
```diff
@@ -52,8 +52,7 @@ public InputSupplier<InputStream> openRoot(String... pathSegments) {
 			return () -> IOUtils.toInputStream(pack, Charsets.UTF_8);
 		} else if ("pack.png".equals(fileName)) {
 			return FabricLoader.getInstance().getModContainer("fabric-resource-loader-v0")
-					.flatMap(container -> container.getMetadata().getIconPath(512).map(container::getPath))
-					.filter(Files::exists)
+					.flatMap(container -> container.getMetadata().getIconPath(512).flatMap(container::findPath))
```
Comment (Technici4n) [inline fabric-resource-loader-v0/src/main/java/net/fabricmc/fabric/impl/resource/loader/FabricModResourcePack.java:55]: Why is the `filter` call not necessary?

**Review by Technici4n — CHANGES_REQUESTED** [review]
Requesting some explanations ! :smile:

File: fabric-resource-loader-v0/src/client/java/net/fabricmc/fabric/mixin/resource/loader/client/VanillaResourcePackProviderMixin.java (hunk 1 above)
Comment (apple502j) [reply fabric-resource-loader-v0/src/client/java/net/fabricmc/fabric/mixin/resource/loader/client/VanillaResourcePackProviderMixin.java:32]: Fixes incorrect mixin injection point, `ServerResourcePackProvider` is used to provide HTTP downloaded packs. This is a better injection point for built in packs.

File: fabric-resource-loader-v0/src/main/java/net/fabricmc/fabric/impl/resource/loader/FabricModResourcePack.java (hunk 2 above)
Comment (apple502j) [reply fabric-resource-loader-v0/src/main/java/net/fabricmc/fabric/impl/resource/loader/FabricModResourcePack.java:55]: See https://maven.fabricmc.net/docs/fabric-loader-0.14.10/net/fabricmc/loader/api/ModContainer.html#findPath(java.lang.String)

**Review by Technici4n — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**